### PR TITLE
WT-12145 Create a lint tag for s-all and clang-analyzer

### PR DESCRIPTION
**Pull request by etienneptl**

### CODE REVIEW

**Review by jiechenbo — APPROVED** [review]
LGTM!

File: test/evergreen.yml [hunk 1]
```diff
@@ -2538,11 +2538,9 @@ tasks:
   # End of Python unit test tasks
 
   - name: s-all
-    tags: ["pull_request"]
```
Comment (lukech) [inline test/evergreen.yml:2541]: Will this change remove this task from the PR testing scope?

Comment (etienneptl) [reply test/evergreen.yml:2541]: Could we update the PR testing to take the new tag?

Comment (lukech) [reply test/evergreen.yml:2541]: Sure, added

**Review by lukech — APPROVED** [review]
LGTM

**Comment by etienneptl** [comment]
evergreen retry

**Comment by etienneptl** [comment]
evergreen retry

**Pull request merged**